src/components/ShoppingList/ShoppingList.js

```diff
@@ -62,6 +62,90 @@ function ShoppingList({ listId }) {
     setFilter(e.target.value);
   };
 
+  const getDaysToPurchase = (item) => {
+    let nextPurchaseDate;
+    if (item.lastPurchaseDate?.seconds) {
+      // if the item has been purchased before, next purchase date is [purchaseInterval] days from the lastPurchaseDate
+      nextPurchaseDate = DateTime.fromSeconds(
+        item.lastPurchaseDate.seconds,
+      ).plus({ days: item.purchaseInterval });
+    } else if (item.createdAt?.seconds) {
+      // if there's no purchase history, estimate it will be bought [purchaseInterval] days from when item was created
+      // (user provides this info at item creation)
+      nextPurchaseDate = DateTime.fromSeconds(item.createdAt.seconds).plus({
+        days: item.purchaseInterval,
+      });
+    } else return null;
```

Comment:
Good catch, because I think nothing? I'm pretty sure that was put in with the intention that we could later check for null when we called the function and handle a situation where there was no lastPurchaseDate or createdAt. But then I don't think we ever added anything in to handle that, so this could be removed.